src/runner/index.js

```diff
@@ -126,6 +137,10 @@ export default class Runner extends EventEmitter {
                 return this._runTask(reporterPlugin, browserSet, tests);
             });
 
-        return Runner._createCancelablePromise(runTaskPromise);
+        return this._createCancelablePromise(runTaskPromise);
+    }
+
+    async stop () {
+        await Promise.all(this.pendingTaskPromises.slice().map(taskPromise => taskPromise.cancel()));
```

Comment:
I have figured out how we can get rid of that nasty `slice()`'s: ES6 `for..of` loop uses Array iterator, that observes changes in array. So it's possible to remove and even insert elements with `splice()` when iterating.